### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by lundibundi**

### CODE REVIEW

File: lib/raw-builder.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const { QueryBuilder } = require('./query-builder');
+
+class RawBuilder extends QueryBuilder {
+  //   sqlTemplate - <Function>
```
Comment (belochub) [inline lib/raw-builder.js]:
```suggestion
  //   sqlTemplate <Function>
```

**Review by belochub — APPROVED** [review]

File: lib/params-builder.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+// Base class for all ParamsBuilders
+class ParamsBuilder {
+  // Adds passed value to parameters
```
Comment (SemenchenkoVitaliy) [inline lib/params-builder.js]:
```suggestion
  // Add passed value to parameters
```

File: lib/pg-params-builder.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+const { ParamsBuilder } = require('./params-builder');
+
+class PostgresParamsBuilder extends ParamsBuilder {
+  constructor() {
+    super();
+    this.params = [];
+  }
+
+  // Adds passed value to parameters
```
Comment (SemenchenkoVitaliy) [inline lib/pg-params-builder.js]:
```suggestion
  // Add passed value to parameters
```

**Review by SemenchenkoVitaliy — APPROVED** [review]

**Review by tshemsedinov — APPROVED** [review]

File: doc/header.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# API documentation
```
Comment (belochub) [inline doc/header.md]: This file can be deleted.

File: lib/params-builder.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+// Base class for all ParamsBuilders
+class ParamsBuilder {
+  // Add passed value to parameters
+  //   value <any>
+  //   options <Object> optional
+  // Returns: <string> name to put in a sql query
```
Comment (belochub) [inline lib/params-builder.js]:
```suggestion
  // Returns: <string> name to put in an SQL query
```

**Review by belochub — APPROVED** [review]

**Pull request merged**